### PR TITLE
Add user, edit user, change role

### PR DESCRIPTION
**Pull request by PawelMiskowiec**
Fixed a small bug in add user and added some js validation.
Added/fixed edit user modal and button in a separate column for it, added some js validation.
Fixed change role functionality.

Added apache HTTP client dependency to support the Patch method in RestTemplate. 
Added exception handling for HttpClientErrorException that passes error responses from GreenCityClient.

### CODE REVIEW

File: core/src/main/java/greencity/exception/handler/HttpClientErrorExceptionResponse.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    @JsonIgnore
+    private String timeStamp;
+    @JsonIgnore
+    private String trace;
+    @JsonIgnore
+    private String path;
+
+    /**
+     * Constructor with parameters.
+     */
```
Comment (marcinkunert) [inline core/src/main/java/greencity/exception/handler/HttpClientErrorExceptionResponse.java]: That's not really a helpful comment 😅

File: core/src/main/java/greencity/webcontroller/ManagementUserController.java [hunk 2]
```diff
@@ -206,9 +207,9 @@ public ResponseEntity<List<String>> getReasonsOfDeactivation(
      * @author Vasyl Zhovnir
      */
     @PostMapping("/deactivateAll")
-    public ResponseEntity<List<Long>> deactivateAll(@RequestBody List<Long> listId) {
-        return ResponseEntity.status(HttpStatus.OK)
-            .body(restClient.deactivateAllUsers(listId));
+    @ResponseStatus(HttpStatus.OK)
```
Comment (marcinkunert) [inline core/src/main/java/greencity/webcontroller/ManagementUserController.java]: Isn't "OK" the default answer?

File: core/src/main/resources/static/management/user/buttonsAJAX.js [hunk 3]
```diff
@@ -106,14 +106,58 @@ function otherCheck() {
...
 
+// Save button in editUserModal
+function updateUser(userId) {
+    clearAllErrorsSpan();
+    let formData = $('#editUserForm' + userId).serializeArray().reduce(function (obj, item) {
+        obj[item.name] = item.value;
+        return obj;
+    }, {});
+    validateForUpdate(formData, userId);
+    let returnData = {
```
Comment (marcinkunert) [inline core/src/main/resources/static/management/user/buttonsAJAX.js]: It's the payload/data, not the return data from server. Consider changing the name

File: core/src/main/resources/static/management/user/buttonsAJAX.js [hunk 4]
```diff
@@ -106,14 +106,58 @@ function otherCheck() {
...
+        }
+    });
+}
+
+function validateForUpdate(formData, userId) {
+    let errorCounter = 0;
+    if(formData.name.length < 6 || formData.name.length > 30){
+        document.getElementById("errorModalUpdateName" + userId).innerText += "Name must contain between 6 and 30 characters";
+        errorCounter++;
+    }
```
Comment (marcinkunert) [inline core/src/main/resources/static/management/user/buttonsAJAX.js]: I would advice addding a more declarative way of performing the validation. Take a look here: https://getbootstrap.com/docs/5.0/forms/validation/

File: service/src/main/java/greencity/service/UserServiceImpl.java [hunk 5]
```diff
@@ -112,13 +113,16 @@ public UserStatusDto updateStatus(Long id, UserStatus userStatus, String email)
...
-        UserVO userVO = findById(id);
-        userVO.setRole(role);
-        userRepo.updateUserRole(id, String.valueOf(role));
-        return modelMapper.map(userVO, UserRoleDto.class);
+        User user = userRepo.findById(id)
+            .orElseThrow(() -> new WrongIdException(ErrorMessage.USER_NOT_FOUND_BY_ID + id));
+        user.setRole(role);
+        userRepo.save(user);
+        return modelMapper.map(user, UserRoleDto.class);
     }
```
Comment (marcinkunert) [inline service/src/main/java/greencity/service/UserServiceImpl.java:132]: Sonsarcube suggests adding @deprecated also to the javadocs, in order to explain what is to be used instaead for other developers.

File: core/src/main/resources/static/management/user/buttonsAJAX.js (hunk 3 above)
Comment (PawelMiskowiec) [reply core/src/main/resources/static/management/user/buttonsAJAX.js]: Oh yeah, I know. I must have messed that up while refactoring

**Review by kmpol — APPROVED** [review]

**Review by lena00-afk — APPROVED** [review]

**Review by F1iper — APPROVED** [review]

**Pull request merged**